test/e2e_tests/utils/capp_adpater.go

```diff
@@ -58,6 +58,18 @@ func GetCapp(k8sClient client.Client, name string, namespace string) *cappv1alph
 	return capp
 }
 
+// UpdateCapp updates the provided Capp instance in the Kubernetes cluster, and returns it.
+func UpdateCapp(k8sClient client.Client, capp *cappv1alpha1.Capp) *cappv1alpha1.Capp {
+	Expect(k8sClient.Update(context.Background(), capp)).To(Succeed())
+
+	Eventually(func() string {
+		updatedCapp := GetCapp(k8sClient, capp.Name, capp.Namespace)
+		return updatedCapp.Status.StateStatus.State
+	}, TimeoutCapp, CappCreationInterval).Should(Equal("enabled"), "Should fetch capp")
+
+	return capp
+}
+
```

Comment:
The `Eventually` here doesn't really work because the `State` is only relevant to test on a first `Capp` creation. On update, in most cases, the State wouldn't change. When doing an `Update` then we should use `Eventually` in a ad-hoc way to see if our change has been made